html/captive-portal/lib/captiveportal/PacketFence/Controller/TLSProfile.pm

```diff
@@ -0,0 +1,180 @@
+package captiveportal::PacketFence::Controller::TLSProfile;
+use Moose;
+use namespace::autoclean;
+use WWW::Curl::Easy;
+use Date::Format qw(time2str);
+use pf::log;
+use pf::config;
+use pf::temporary_password 1.11;
+use pf::util;
+use pf::web qw(i18n ni18n i18n_format render_template);
+use pf::web::constants;
+use pf::web::util;
+use pf::web::guest;
+use pf::activation;
+use pf::Authentication::constants;
+use pf::Authentication::Action;
+use pf::authentication;
+use List::MoreUtils qw(uniq any);
+use Readonly;
+use POSIX;
+use URI::Escape::XS qw(uri_escape);
+use pf::web;
+use lib qw(/usr/local/pf/lib);
+
+
+BEGIN { extends 'captiveportal::Base::Controller'; }
+
+=head1 NAME
+
+captiveportal::PacketFence::Controller::WirelessProfile - Catalyst Controller
+
+=head1 DESCRIPTION
+
+Catalyst Controller.
+
+=head1 METHODS
+
+=cut
+
+=head2 index
+
+=cut
+
+sub index : Path : Args(0) {
+    my ( $self, $c ) = @_;
+    my $username = $c->session->{username};
+    my $logger  = get_logger;
+    my $profile = $c->profile;
+    my $request = $c->request;
+    my $mac = $c->portalSession->clientMac;
+    my $provisioner = $c->profile->findProvisioner($mac);
+    $provisioner->authorize($mac) if (defined($provisioner));
```

Comment:
What is the point of calling authorize here, you're not doing anything with what it returns you.